playground/backend/internal/storage/cloud_storage.go

```diff
@@ -0,0 +1,218 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api/v1"
+	"beam.apache.org/playground/backend/internal/logger"
+	"cloud.google.com/go/storage"
+	"context"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	BucketName      = "playground-examples"
+	OutputExtension = "output"
+	MetaInfoName    = "meta.info"
+	Timeout         = 10
+	javaExtension   = "java"
+	goExtension     = "go"
+	pyExtension     = "py"
+	scioExtension   = "scala"
+)
+
+type ExamplesInfo []pb.Example
+type SdkCategories map[string]ExamplesInfo
+type Examples map[string]SdkCategories
```

Comment:
Does it contain SDK as key and categories as value? If yes, why is it named `examples`?

Reply:
Oh, each SdkCategories contains examples? Very implicit naming. It's really easy to name better for understanding. SdkCategories -> CategoriesWithExamples.